Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean

```diff
@@ -76,6 +76,20 @@ We use the following type variables in this file:
 
 universe u v v' wE wE₁ wE' wG wG'
 
+/-- Applying a multilinear map to a vector is continuous in both coordinates. -/
+theorem ContinuousMultilinearMap.continuous_eval {𝕜 ι : Type*} {E : ι → Type*} {F : Type*}
+    [NormedField 𝕜] [Finite ι] [∀ i, SeminormedAddCommGroup (E i)] [∀ i, NormedSpace 𝕜 (E i)]
+    [TopologicalSpace F] [AddCommGroup F] [TopologicalAddGroup F] [Module 𝕜 F] :
+    Continuous fun p : ContinuousMultilinearMap 𝕜 E F × ∀ i, E i => p.1 p.2 := by
+  cases nonempty_fintype ι
+  letI := TopologicalAddGroup.toUniformSpace F
+  haveI := comm_topologicalAddGroup_is_uniform (G := F)
```

Comment:
```suggestion
  let := TopologicalAddGroup.toUniformSpace F
  have := comm_topologicalAddGroup_is_uniform (G := F)
```
Are `letI` and `haveI` still useful in Lean 4?

Reply:
AFAIR, I used `letI` to avoid `_`.

Reply:
They are useful, if you want to inline your `let`.